rig/place_and_route/utils.py

```diff
@@ -136,3 +141,18 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
             )
 
     return routing_tables
+
+
+class MultisourceRouteError(Exception):
+    """Indicates that two nets with the same key and mask would cause packets
+    to become duplicated.
+    """
+    def __init__(self, key, mask, coordinate):
+        self.key = key
+        self.mask = mask
+        self.x, self.y = coordinate
+
+    def __str__(self):
+        return ("Two different nets with the same key ({0.key:#010x}) and "
+                "mask ({0.mask:#010x}) fork differently at ({0.x}, {0.y})".
```

Comment:
By the way, it may be worth ending the exception with "This is almost never what you want" or something. Maybe this check should be an optional feature (e.g. to allow redundant entries to be inserted which will later get "unmasked")?